string[] nodes
pgm_learner/GraphEdge[] edges
